src/tv2-common/actions/executeAction.ts

```diff
@@ -1053,34 +1045,57 @@ async function executeActionCutToCamera<
 		}
 	}
 
-	if (userData.queue || serverInCurrentPart) {
-		await context.core.queuePart(part, [
-			kamPiece,
+	await executePiece(context, settings, kamPiece, userData.queue, part)
+}
+
+async function executePiece<
+	StudioConfig extends TV2StudioConfigBase,
+	ShowStyleConfig extends TV2BlueprintConfigBase<StudioConfig>
+>(
+	context: ActionExecutionContext<ShowStyleConfig>,
+	settings: ActionExecutionSettings<StudioConfig, ShowStyleConfig>,
+	pieceToExecute: IBlueprintPiece<PieceMetaData>,
+	shouldBeQueued: boolean,
+	partToQueue: IBlueprintPart
+) {
+	const currentPieceInstances = await context.core.getPieceInstances('current')
+	const serverInCurrentPart = currentPieceInstances.some(
```

Comment:
When I read this I expect the `serverInCurrentPart` to be a Part describing a server clip, but the value is a boolean. I'll suggest renaming it to `isServerInCurrentPart`.

Reply:
Updated